### PR TITLE
protopuf: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **protopuf/2.2.0**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
I took 10 minutes to understand that this project is protopuf and not protobuf

File: recipes/protopuf/all/test_v1_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 3.12)
```
Comment (uilianries) [inline recipes/protopuf/all/test_v1_package/CMakeLists.txt:1]: good catch about cxx_std_20

File: recipes/protopuf/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.layout import basic_layout
+import os
+
+
+required_conan_version = ">=1.52.0"
```
Comment (uilianries) [inline recipes/protopuf/all/conanfile.py:10]:
```suggestion
required_conan_version = ">=1.47.0"
```
1.47.0 should be enough here

File: recipes/protopuf/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy
```
Comment (uilianries) [inline recipes/protopuf/all/conanfile.py:3]:
```suggestion
from conan.tools.files import get, copy
```
not used

**Review by uilianries — COMMENTED** [review]
Good, only few points

File: recipes/protopuf/all/conanfile.py (hunk 3 above)
Comment (toge) [reply recipes/protopuf/all/conanfile.py:3]: remove unused functions

File: recipes/protopuf/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            "msvc": "192",
+            "gcc": "10",
+            "clang": "12",
+        }
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def package_id(self):
+        self.info.clear()
```
Comment (jwillikers) [inline recipes/protopuf/all/conanfile.py:40]: This requires Conan 1.50.0: https://docs.conan.io/en/latest/changelog.html#jun-2022

File: recipes/protopuf/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import get, copy
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.layout import basic_layout
+import os
+
+
+required_conan_version = ">=1.47.0"
```
Comment (jwillikers) [inline recipes/protopuf/all/conanfile.py:10]:
```suggestion
required_conan_version = ">=1.50.0"
```

File: recipes/protopuf/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler == "apple-clang":
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._minimum_cpp_standard}, apple-clang does not support(yet)."
```
Comment (jwillikers) [inline recipes/protopuf/all/conanfile.py:45]:
```suggestion
                f"{self.ref} does not yet support apple-clang."
```

File: recipes/protopuf/all/test_package/CMakeUserPresets.json [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (jwillikers) [inline recipes/protopuf/all/test_package/CMakeUserPresets.json:1]: This file should not be added to version control. We must need to add this to the `.gitignore`.

**Review by jwillikers — CHANGES_REQUESTED** [review]

File: recipes/protopuf/all/test_package/CMakeUserPresets.json (hunk 7 above)
Comment (jwillikers) [reply recipes/protopuf/all/test_package/CMakeUserPresets.json:1]: I've submitted a PR to add this to the `.gitignore`: #13348

Comment (toge) [reply recipes/protopuf/all/test_package/CMakeUserPresets.json]: @jwillikers 
Thanks!
I removed CMakeUserPresets.json from this PR.

**Review by jwillikers — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**